src/main/osd/osd_elements.c

```diff
@@ -156,6 +157,45 @@ static uint32_t blinkBits[(OSD_ITEM_COUNT + 31) / 32];
 #define IS_BLINK(item) (blinkBits[(item) / 32] & (1 << ((item) % 32)))
 #define BLINK(item) (IS_BLINK(item) && blinkState)
 
+#if defined(USE_ESC_SENSOR) || defined(USE_RPM_FILTER)
+typedef int (*getEscRpmOrFreqFnPtr)(int i);
+
+static int getEscRpm(int i)
+{
+#ifdef USE_RPM_FILTER
+    if (motorConfig()->dev.useDshotTelemetry) {
+        return 100.0f / (motorConfig()->motorPoleCount / 2.0f) * getDshotTelemetry(i);
+    }
+#endif 
+    if (featureIsEnabled(FEATURE_ESC_SENSOR)) {
+        return calcEscRpm(getEscSensorData(i)->rpm);
+    } else { 
+        return 0;
+    }
+}
+
+static int getEscRpmFreq(int i) 
+{
+    return getEscRpm(i) / 60;
+}
+
+static void renderOsdEscRpmOrFreq(getEscRpmOrFreqFnPtr escFnPtr, osdElementParms_t *element)
+{
+    int x = element->elemPosX;
+    int y = element->elemPosY;
+    for (int i=0; i < getMotorCount(); i++) {
+        char rpmStr[6];
+        int rpm = (*escFnPtr)(i);
+        if ( rpm > 99999 ) {  // avoid buffer overflow
+            rpm = 99999;
+        }
+        int len = tfp_sprintf(rpmStr, "%d", rpm);
+        rpmStr[len] = '\0'; 
+        displayWrite(element->osdDisplayPort, x, y + i, rpmStr);
+        element->drawElement = false;
```

Comment:
This shouldn't be in the `for` loop. It just needs to be set once for the entire element.

Reply:
yes, that is very true,  it is fixed now.